src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averagegrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAvGrade(grades: number[]): number {
+  return grades.reduce((sum, element) => sum + element, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+):Student[] {
+  const classroom = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? classroom.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
+        : classroom.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));
+
+    case SortType.Age:
+    case SortType.Married:
+      return order === 'asc'
+        ? classroom.sort((a, b) => Number(a[sortBy]) - Number(b[sortBy]))
+        : classroom.sort((a, b) => Number(b[sortBy]) - Number(a[sortBy]));
+
+    case SortType.AverageGrade:
+      return order === 'asc'
+        ? classroom.sort((a, b) => getAvGrade(a.grades) - getAvGrade(b.grades))
+        : classroom.sort((a, b) => getAvGrade(b.grades) - getAvGrade(a.grades));
```

Comment:
i just had a thought:
```suggestion
	classroom.sort((a, b) => {
	  switch (sortBy) {
     case SortType.Name:
     case SortType.Surname:
       return a[sortBy].localeCompare(b[sortBy]);

      case SortType.Age:
      case SortType.Married:
    	  return Number(a[sortBy]) - Number(b[sortBy]);

	    case SortType.AverageGrade:
  	    return classroom.sort((a, b) => getAvGrade(a.grades) - getAvGrade(b.grades));
		}
	}
  
  return order === 'asc'
  	? classroom
  	: classroom.reverse();
```